dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/ToolTest.java

```diff
@@ -931,4 +931,327 @@ public void testToggleWithBooleanArgs() throws Exception {
+
+        String response = HttpTestUtils.callMCP(server, "/toolTest", request);
+        String expectedResponseString = """
+                        {"id":"2","jsonrpc":"2.0","result":{"content":[{"type":"text","text":"c"}], "isError": false}}
+                        """;
+        JSONAssert.assertEquals(expectedResponseString, response, true);
+    }
+
+    @Test
+    public void testJSONlong() throws Exception {
+        String request = """
+                          {
+                          "jsonrpc": "2.0",
+                          "id": "2",
+                          "method": "tools/call",
+                          "params": {
+                            "name": "testJSONlong",
+                            "arguments": {
+                              "num1": 2
+                            }
+                          }
+                        }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, "/toolTest", request);
+        String expectedResponseString = """
+                        {"id":"2","jsonrpc":"2.0","result":{"content":[{"type":"text","text":2}],"isError":false}}
+                        """;
+        JSONAssert.assertEquals(expectedResponseString, response, true);
+    }
+
+    @Test
+    public void testJSONdouble() throws Exception {
+        String request = """
+                          {
+                          "jsonrpc": "2.0",
+                          "id": "2",
+                          "method": "tools/call",
+                          "params": {
+                            "name": "testJSONdouble",
+                            "arguments": {
+                              "num1": 2
+                            }
+                          }
+                        }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, "/toolTest", request);
+        String expectedResponseString = """
+                        {"id":"2","jsonrpc":"2.0","result":{"content":[{"type":"text","text":2}], "isError": false}}
+                        """;
+        JSONAssert.assertEquals(expectedResponseString, response, true);
+    }
+
+    @Test
+    public void testJSONbyte() throws Exception {
+        String request = """
+                          {
+                          "jsonrpc": "2.0",
+                          "id": "2",
+                          "method": "tools/call",
+                          "params": {
+                            "name": "testJSONbyte",
+                            "arguments": {
+                              "num1": 2
+                            }
+                          }
+                        }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, "/toolTest", request);
+        String expectedResponseString = """
+                        {"id":"2","jsonrpc":"2.0","result":{"content":[{"type":"text","text":2}], "isError": false}}
+                        """;
```

Comment:
Unintuitive as it is here where the response is a number, if we're returning it as "text" content (which I think is the only sensible option), we need to put a string in the result, not a number.

```suggestion
        String expectedResponseString = """
                        {"id":"2","jsonrpc":"2.0","result":{"content":[{"type":"text","text":"2"}], "isError": false}}
                        """;
```

Schema reference: https://modelcontextprotocol.io/specification/2025-06-18/schema#textcontent

Reply:
This needs to change for all the tests. Where we have `"content":[{"type":"text","text": <<something>>}]`, the `<<something>>` must be a string.

Reply:
FYI: The number tests will all fail if I make the expected response numbers a string like this: "2"

Reply:
Yeah, that's expected, we need to change the implementation code as well to ensure that it returns it as a string.

Correct the tests first, and then we can work on fixing the code to make the tests pass.

Reply:
OK.